frontend/src/metabase-lib/lib/Mode/utils.ts

```diff
@@ -1,35 +1,40 @@
-import SegmentMode from "../components/modes/SegmentMode";
-import MetricMode from "../components/modes/MetricMode";
-import TimeseriesMode from "../components/modes/TimeseriesMode";
-import GeoMode from "../components/modes/GeoMode";
-import PivotMode from "../components/modes/PivotMode";
-import NativeMode from "../components/modes/NativeMode";
-import DefaultMode from "../components/modes/DefaultMode";
-
+import Question from "metabase-lib/lib/Question";
 import StructuredQuery from "metabase-lib/lib/queries/StructuredQuery";
 import NativeQuery from "metabase-lib/lib/queries/NativeQuery";
+import { ModeType } from "./types";
+import {
+  MODE_TYPE_NATIVE,
+  MODE_TYPE_SEGMENT,
+  MODE_TYPE_METRIC,
+  MODE_TYPE_TIMESERIES,
+  MODE_TYPE_GEO,
+  MODE_TYPE_PIVOT,
+  MODE_TYPE_DEFAULT,
+} from "metabase-lib/lib/Mode/constants";
 
-export function getMode(question) {
+export function getMode(question: Question): ModeType | null {
```

Comment:
Returns the mode type only, thus making it independent from `metabase-types/types/Visualization` and `frontend/src/metabase/modes`.